src/reconcile.js

```diff
@@ -0,0 +1,266 @@
+
+module.exports = function (clusterInfo, uuid = require('uuid')) {
+  return {
+    processArgv,
+    storeResult,
+
+    // for testing
+    _reconcileName,
+    _yargsOverrides,
+    _reifyServiceAccounts
+  }
+
+  /**
+   * A lot is going on here!  We have to take in the yargs args, and then
+   * produce the params needed for fabrik8.  Some of these will be inferred
+   * from the input, others will come from Vault as default values.  Some params
+   * will come from previous fabrik8 runs, if the cluster exists!  We also have
+   * to process args that override defaults, and generate certain missing values
+   * according to rules specified in the common default config.
```

Comment:
having all of this in one place is _awesome_ so thank you for that

i'd love to have some kind of documentation as to what all of the parameters are and what sources they can come from, as well as how other sources fit in order of precedence, but that is very much not in scope for this pull request and should not be a blocker